### PR TITLE
chore(sdk): fix type annotation for wandb.Api to Type[PublicApi]

### PR DESCRIPTION
**Pull request by bdvllrs**
# Description
What does the PR do? Include a concise description of the PR contents.

- Fixes: #8520

The type changes in #8447 introduced a type error for `wandb.Api` from `Type[PublicApi]` to `PublicApi`.

### CODE REVIEW

File: wandb/__init__.template.pyi [hunk 1]
```diff
@@ -100,7 +101,7 @@ __version__: str = "0.18.4.dev1"
 run: Run | None
 config: wandb_config.Config
 summary: wandb_summary.Summary
-Api: PublicApi
+Api: Type[PublicApi]
```
Comment (kptkin) [inline wandb/__init__.template.pyi:104]: we use `from __future__ import annotations`, so i believe you can just use `type[PublicApi]`. also if you don't mind fixing the config/summary that would be awesome :)

**Review by kptkin — APPROVED** [review]
Thanks a lot for putting up a fix!

**Comment by kptkin** [comment]
Another small request: if you could update the CHANGELOG here: https://github.com/wandb/wandb/blob/main/CHANGELOG.md#unreleased 
adding your change, so it would be included in the release notes and you get attribution for the changes you made

File: wandb/__init__.template.pyi (hunk 1 above)
Comment (bdvllrs) [reply wandb/__init__.template.pyi:104]: Yes, no problem

**Comment by bdvllrs** [comment]
> Another small request: if you could update the CHANGELOG here: https://github.com/wandb/wandb/blob/main/CHANGELOG.md#unreleased adding your change, so it would be included in the release notes and you get attribution for the changes you made

Yes, I added a line to the changelog.

File: wandb/__init__.pyi [hunk 2]
```diff
@@ -100,7 +100,7 @@ __version__: str = "0.18.4.dev1"
 run: Run | None
 config: wandb_config.Config
 summary: wandb_summary.Summary
-Api: PublicApi
+PUBLIC_REPO: type[PublicApi]
```
Comment (kptkin) [inline wandb/__init__.pyi:103]:
```suggestion
Api: type[PublicApi]
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +24,7 @@ Please add to the relevant subsections under Unreleased below on every PR where
 - `run.log_artifact()` no longer blocks other data uploads until the artifact upload finishes (@timoffex in https://github.com/wandb/wandb/pull/8466)
 - Fixed media dependency for rdkit updated from `rdkit-pypi` to `rdkit` (@jacobromero in https://github.com/wandb/wandb/compare/WB-20894)
 - Saving an artifact with many large files no longer exhausts OS threads (@timoffex in https://github.com/wandb/wandb/pull/8518)
+- Fixed typing issue of `wandb.Api` (@bdvllrs in https://github.com/wandb/wandb/pull/8548)
```
[the remark this answers is not in the document]
Comment (bdvllrs) [reply CHANGELOG.md]: Yes, I didn't notice as the Unreleased was empty. Should be good now!

**Review by kptkin — APPROVED** [review]

**Pull request merged**